src/safir/middleware.py

```diff
@@ -0,0 +1,140 @@
+"""Middleware for aiohttp.web servers.
+"""
+
+__all__ = ["bind_logger"]
+
+import uuid
+from typing import TYPE_CHECKING, Awaitable, Callable
+
+import structlog
+from aiohttp import web
+
+from safir.logging import response_logger
+
+if TYPE_CHECKING:
+    from aiohttp.web.web_response import Request, StreamResponse
+
+_Handler = Callable[["Request"], Awaitable["StreamResponse"]]
+_Middleware = Callable[["Request", _Handler], Awaitable["StreamResponse"]]
```

Comment:
I generally move these into the `if TYPE_CHECKING` block since they're only used in type annotations. That also lets you move `Awaitable` and `Callable` into the `if TYPE_CHECKING` block.